File: src/commands/git/AddDot.ts
import { Option } from '../../common/interface/Option';
import { SuperCommand } from '../../common/SuperCommand';
import { executeCommand } from '../../common/Cli';
import { Argument } from '../../common/interface/Argument';

const argumentss: Argument[] = [];
const options: Option[] = [];

export class Command extends SuperCommand {
  description: string = 'Stages all files and folders in the current directory';
  alias: string = 'gad';
  public async execute(vital?: boolean, input?: any) {
    await super.executeWithInput(argumentss, options, input, vital, async () => {
      await executeCommand('git', ['add', '.']);
    });
  }
}

File: src/workflow_shell/commands/git/add_dot.clj
(ns workflow-shell.commands.git.add_dot
  (:require [clojure.java.shell :refer [sh]])
  (:gen-class))

(def gad (hash-map
         :description "Stages all files and folders in the current directory"
         :execute (fn [& args]
                    (let [result  (sh "git" "add" ".")]
                      (if-not (empty? (:out result))
                        (println (:out result))
                        (println (:err result)))))))